### PR TITLE
docs: patch hieroglyph to work with sphinx 1.8.0+

### PR DESCRIPTION
**Pull request by oliver-sanders**
Temporary fix to allow the documentation to be built with contemporary versions of Sphinx (note v2.0 is under development).

### CODE REVIEW

File: bin/rose-make-docs [hunk 1]
```diff
@@ -179,10 +179,10 @@ venv-install () {
     venv-activate
     # TODO remove Sphinx version dependency
```
Comment (sadielbartholomew) [inline bin/rose-make-docs]: This TODO has just been addressed, so can be removed I think (:ballot_box_with_check:)?

File: sphinx/ext/hieroglyph_patch.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+from sphinx.writers.html import HTMLTranslator
+
+from sphinx.ext.graphviz import html_visit_graphviz
+from sphinx.ext.autosummary import (autosummary_toc_visit_html,
+                                    autosummary_table_visit_html)
+from minicylc import MiniCylc
+
+
+def none(*args, **kwargs):
```
Comment (sadielbartholomew) [inline sphinx/ext/hieroglyph_patch.py]: Is this just left from development or does it have some mysterious purpose?

Comment (oliver-sanders) [reply sphinx/ext/hieroglyph_patch.py]: It's a dummy interface used below on lines 36 and 38.

**Comment by oliver-sanders** [comment]
> Is that just something to do with my environment or setup?

Yes, make sure you are viewing the site through a web server (e.g. Apache) rather than viewing the file directly in the browser.

Explaination, the version selector make an AJAX call to load in a JSON file which contains the information about which versions and formats are available.

File: sphinx/ext/hieroglyph_patch.py (hunk 2 above)
Comment (sadielbartholomew) [reply sphinx/ext/hieroglyph_patch.py]: Oh, sorry, I did check for that but missed them (can't have looked very well!)

**Pull request merged**